dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperTransporter.java

```diff
@@ -137,8 +138,8 @@ void writeToClientMap(List<String> addressList, ZookeeperClient zookeeperClient)
     URL toClientURL(URL url) {
         Map<String, String> parameterMap = new HashMap<>();
         // for CuratorZookeeperClient
-        if (url.getParameter(TIMEOUT_KEY) != null) {
-            parameterMap.put(TIMEOUT_KEY, url.getParameter(TIMEOUT_KEY));
+        if (url.getParameter(Constants.CONFIG_TIMEOUT_KEY) != null) {
+            parameterMap.put(Constants.CONFIG_TIMEOUT_KEY, url.getParameter(Constants.CONFIG_TIMEOUT_KEY));
```

Comment:
This is not right, because the zookeeper transporter is not only used by config-center. It will not use "config.timeout" as timeout key when it is used by Zookeeper Registry.
And it seems your pr is not clean, you should rebase your repository from official upstream master branch.
I have send a pr #4763 to solve the problem. (I am sorry that I didn't see your pr when I sent my pr. )